modules/service/src/main/java/org/mule/runtime/module/service/api/artifact/ServiceClassLoaderFactory.java

```diff
@@ -53,6 +54,25 @@ public ArtifactClassLoader create(String artifactId, ServiceDescriptor descripto
                                        containerClassLoader.getContainerClassLoaderLookupPolicy());
   }
 
+  /**
+   * Creates a {@link ClassLoader} from a given descriptor.
+   *
+   * @param artifactId           artifact unique ID.
+   * @param descriptor           descriptor of the artifact owner of the created class loader.
+   * @param containerClassLoader parent for the new artifact class loader.
+   * @return a new class loader for described artifact.
+   * 
+   * @since 4.6
+   */
+  public ArtifactClassLoader create(String artifactId, ServiceDescriptor descriptor,
+                                    ArtifactClassLoader containerClassLoader) {
+    final MuleContainerClassLoaderWrapper containerClassLoaderWrapper =
+        new DefaultMuleContainerClassLoaderWrapper(containerClassLoader);
+    return new MuleArtifactClassLoader(artifactId, descriptor, descriptor.getClassLoaderConfiguration().getUrls(),
+                                       containerClassLoaderWrapper.getContainerClassLoader().getClassLoader(),
+                                       containerClassLoaderWrapper.getContainerClassLoaderLookupPolicy());
```

Comment:
Why not delegating directly on the constructor `create(String artifactId, ServiceDescriptor descriptor, MuleContainerClassLoaderWrapper containerClassLoader)`?